File: app/src/main/java/com/jugger/tetris/Basic/Brick.java
package com.jugger.tetris.Basic;

public class Brick {

    private Pos p;

    private int c;

    private int image;

    public Brick(Pos p, int c) {
        this.p = p;
        this.c = c;
    }

    public Brick(Pos p, int c, int image) {
        this.p = p;
        this.c = c;
        this.image = image;
    }

    public Brick copy() {
        return new Brick(p.copy(), c, image);
    }

    public void changePos(int r, int c) {
        p = new Pos(p.r + r, p.c + c);
    }

    public int getImage() {
        return image;
    }

    public void setImage(int image) {
        this.image = image;
    }

    public int getColor() {
        return c;
    }

    public Pos getPosition() {
        return p;
    }
}

File: app/src/main/java/com/jugger/tetris/Basic/Brick.kt
package com.jugger.tetris.Basic

import android.graphics.Color

open class Brick {
    var point: Point = Point(-1,-1)
        private set
    var color = Color.GRAY
        private set
    var image = -1

    companion object {
        val emptyBrick_grey = EmptyBrick(  Color.GRAY)
        val emptyBrick_white = EmptyBrick( Color.WHITE)
    }

    class EmptyBrick: Brick {
        constructor( color:Int) : super( color) {
        }
    }


    constructor(p: Point, c: Int, image: Int) {
        point = p
        color = c
        this.image = image
    }

    constructor( color:Int) {
        this.color = color
    }

    fun copy(): Brick {
        return Brick(point.copy(), color, image)
    }

    fun changePos(r: Int, c: Int) {
        point = Point(point.r + r, point.c + c)
    }

}